pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -246,73 +248,110 @@ func (e *EbpfProgram) InitWithLibsets(libsets ...Libset) error {
 
 	e.setupManagerAndPerfHandlers()
 
-	if err := e.loadProgram(libsets); err != nil {
+	// Mark the libsets as requested so they can be started
+	// Note that other libsets might be requested from previous executions
+	for _, libset := range libsets {
+		e.libsets[libset].requested = true
+	}
```

Comment:
If a libset does not exist in `e.Libsets` we will panic.
We should verify the libsest exists, and if it is not, then to return an error
If we can have a better way to ensure it cannot happen - that's fine by me as well

Reply:
I changed the validation at the start to use `e.isLibsetValid`, that way the "validity" check is centralized in the initialization of the `libsets` map, and we avoid panics in case that changes.